WooCommerce/WooCommerceUITests/Screens/Login/PrologueScreen.swift

```diff
@@ -1,20 +1,27 @@
-import Foundation
+import ScreenObject
 import XCTest
 
 private struct ElementStringIDs {
     static let continueButton = "Prologue Continue Button"
     static let siteAddressButton = "Prologue Self Hosted Button"
 }
 
-final class PrologueScreen: BaseScreen {
-    private let continueButton: XCUIElement
-    private let siteAddressButton: XCUIElement
+final class PrologueScreen: ScreenObject {
+    private let continueButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons[ElementStringIDs.continueButton]
+    }
+    private let siteAddressButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons[ElementStringIDs.siteAddressButton]
+    }
 
-    init() {
-        continueButton = XCUIApplication().buttons[ElementStringIDs.continueButton]
-        siteAddressButton = XCUIApplication().buttons[ElementStringIDs.siteAddressButton]
+    private var continueButton: XCUIElement { expectedElement }
+    private var siteAddressButton: XCUIElement { siteAddressButtonGetter(app) }
 
-        super.init(element: continueButton)
+    init(app: XCUIApplication = XCUIApplication()) throws {
+        try super.init(
+            expectedElementGetter: { $0.buttons[ElementStringIDs.continueButton] },
+            app: app
```

Comment:
This is more of a comment about `ScreenObject` that this PR, but the two sort of go hand in hand since the PR exercises this framework that is still in flux.

Here I'm passing a closure to get the element from the app and the app itself because I felt uncomfortable about the possibility of passing an element from one app and then a different app instance. By using the closure instead, we can ensure that `ScreenObject` will look for the element in the same app it stores.

Now that I say this all out loud, I do wonder whether `ScreenObject` even needs to store an `app` instance? It makes sense for its subclasses because they may need to access multiple elements, but maybe the base class can get away without it?

Reply:
IMHO it doesn't hurt to have it since subclasses are likely to need it? Also should help folks avoid using globals to access the app instance all the time?

Reply:
Very good point. I came to a similar conclusion eventually. `ScreenObject` currently is [storing and exposing `app` as a `let`](https://github.com/Automattic/ScreenObject/blob/trunk/Sources/ScreenObject/ScreenObject.swift#L9-L11).